bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,337 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result, RESULT_SEVERITY
+from coalib.results.Diff import Diff
+
+
+def generate_spacing_diff(file, filename, line, line_number,
+                          match_object, required_spacing):
+    '''
+    Generate a diff for incorrectly spaced control or variable tags.
+
+    :param match_object:     A Match object containing the groups ``open``,
+                             ``close`` containing the opening and closing
+                             delimiters of a Jinja2 tag and ``content``
+                             containing everything in between.
+    :param required_spacing: The number of spaces expected after the ``open``
+                             delimiter and before the ``close`` delimiter
+    '''
+    diff = Diff(file)
+
+    content_before = line[:match_object.start("open")]
+    content_after = line[match_object.end("close"):]
+
+    spacing = " " * required_spacing
+    replacement = (
+        "{before}{open}{spacing}{content}{spacing}{close}{after}".format(
+            before=content_before,
+            spacing=spacing,
+            after=content_after,
+            content=match_object.group("content").strip(),
+            open=match_object.group("open"),
+            close=match_object.group("close")))
+    diff.change_line(
+        line_number,
+        line,
+        replacement)
```

Comment:
The function will only be called if there is an issue, i.e. the line is not as expected. Therefore replacement == line can't happen